src/system.h

```diff
@@ -783,19 +783,54 @@ extern void SyAbortBags(const Char * msg) NORETURN;
 /****************************************************************************
 **
 *F * * * * * * * * * * * * * loading of modules * * * * * * * * * * * * * * *
+**
+** GAP_KERNEL_API_VERSION gives the version of the GAP kernel. This value
+** is used to check if kernel modules were built with a compatible kernel.
+** This version is not the same as, and not connected to, the GAP version.
+**
+** The kernel will not compile a version is stored similarly to 'semver',
+** as major*1000+minor.
+**
+** The algorithm used is the following:
+**
+** The kernel will not load a module compiled for a newer kernel.
+**
+** The kernel will not load a module compiled for a different major version.
+**
+** The minor version should be incremented when new backwards-compatible
+** functionality is added. The major version should be incremented when
+** a backwards-incompatible change is made.
+**
 */
 
+enum { GAP_KERNEL_API_VERSION = 1001 };
+
 enum {
     /** builtin module */
-    MODULE_BUILTIN = 1,
+    MODULE_BUILTIN = GAP_KERNEL_API_VERSION * 10,
 
     /** statically loaded compiled module */
-    MODULE_STATIC  = 2,
+    MODULE_STATIC = GAP_KERNEL_API_VERSION * 10 + 1,
 
     /** dynamically loaded compiled module */
-    MODULE_DYNAMIC = 3,
+    MODULE_DYNAMIC = GAP_KERNEL_API_VERSION * 10 + 2,
```

Comment:
Why use 10, and not 3 or 4 or 256? Perhaps to help during debugging? (I am neutral on that, just wondering)

Reply:
Note that other projects often use hex representation for the same purpose. But again, I am completely agnostic, as I can use `print val/4` or `print val/10` equally well in lldb and gdb :-).

Reply:
Seeing as these are UInts, when printed in lldb they would come out as base-10 numbers, so base-10 seems like the easier thing to use, as you say so when printed out it is readable!